crates/bevy_core_pipeline/src/fxaa/node.rs

```diff
@@ -0,0 +1,132 @@
+use std::sync::Mutex;
+
+use crate::fxaa::{CameraFxaaPipeline, Fxaa, FxaaPipeline};
+use bevy_ecs::prelude::*;
+use bevy_ecs::query::QueryState;
+use bevy_render::{
+    render_graph::{Node, NodeRunError, RenderGraphContext, SlotInfo, SlotType},
+    render_resource::{
+        BindGroup, BindGroupDescriptor, BindGroupEntry, BindingResource, FilterMode, Operations,
+        PipelineCache, RenderPassColorAttachment, RenderPassDescriptor, SamplerDescriptor,
+        TextureViewId,
+    },
+    renderer::RenderContext,
+    view::{ExtractedView, ViewTarget},
+};
+use bevy_utils::default;
+
+pub struct FxaaNode {
+    query: QueryState<
+        (
+            &'static ViewTarget,
+            &'static CameraFxaaPipeline,
+            &'static Fxaa,
+        ),
+        With<ExtractedView>,
+    >,
+    cached_texture_bind_group: Mutex<Option<(TextureViewId, BindGroup)>>,
+}
+
+impl FxaaNode {
+    pub const IN_VIEW: &'static str = "view";
+
+    pub fn new(world: &mut World) -> Self {
+        Self {
+            query: QueryState::new(world),
+            cached_texture_bind_group: Mutex::new(None),
+        }
+    }
+}
+
+impl Node for FxaaNode {
+    fn input(&self) -> Vec<SlotInfo> {
+        vec![SlotInfo::new(FxaaNode::IN_VIEW, SlotType::Entity)]
+    }
+
+    fn update(&mut self, world: &mut World) {
+        self.query.update_archetypes(world);
+    }
+
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let view_entity = graph.get_input_entity(Self::IN_VIEW)?;
+        let pipeline_cache = world.resource::<PipelineCache>();
+        let fxaa_pipeline = world.resource::<FxaaPipeline>();
+
+        let (target, pipeline, fxaa) = match self.query.get_manual(world, view_entity) {
+            Ok(result) => result,
+            Err(_) => return Ok(()),
+        };
+
+        if !fxaa.enabled {
+            return Ok(());
+        };
+
+        let pipeline = pipeline_cache
+            .get_render_pipeline(pipeline.pipeline_id)
+            .unwrap();
+
+        let post_process = target.post_process_write();
+        let source = post_process.source;
+        let destination = post_process.destination;
```

Comment:
Is there a reason for this or is it just to have shorter variable names?

Reply:
As far as I understand, yes.

Reply:
Yup just that